src/settings/global_user.rs

```diff
@@ -11,18 +12,37 @@ impl GlobalUser {
     pub fn new() -> Result<Self, failure::Error> {
         get_global_config()
     }
+    pub fn config_directory() -> Option<PathBuf> {
+        get_global_config_directory()
+    }
+}
+
+fn get_global_config_directory() -> Option<PathBuf> {
+    let deprecated_directory = dirs::home_dir()
+        .expect("oops no home dir")
+        .join(".wrangler")
+        .join("config");
+
+    let directory;
+    if deprecated_directory.exists() {
+        directory = Some(deprecated_directory);
+    } else {
+        directory = dirs::config_dir().map(|p| p.join("wrangler"));
```

Comment:
For debbuing it would be useful to add `info!` in these branches. Something like:
```rust
        info!("using legacy directory {:?}", directory);
```
and
```rust
        info!("using xdg directory {:?}", directory);
```